crates/relayer/src/keyring/ed25519_key_pair.rs

```diff
@@ -0,0 +1,120 @@
+use core::any::Any;
+
+use bip39::{Language, Mnemonic, Seed};
+use ed25519_dalek::{Keypair, PublicKey, SecretKey};
+use ed25519_dalek_bip32::{ChildIndex, DerivationPath, ExtendedSecretKey};
+use hdpath::StandardHDPath;
+use serde::{Deserialize, Serialize};
+use signature::Signer;
+
+use super::{errors::Error, KeyFile, KeyType, SigningKeyPair};
+use crate::config::AddressType;
+
+pub fn private_key_from_mnemonic(
+    mnemonic_words: &str,
+    hd_path: &StandardHDPath,
+) -> Result<ExtendedSecretKey, Error> {
+    let mnemonic = Mnemonic::from_phrase(mnemonic_words, Language::English)
+        .map_err(Error::invalid_mnemonic)?;
+
+    let seed = Seed::new(&mnemonic, "");
+
+    let base_key = ExtendedSecretKey::from_seed(seed.as_bytes())
+        .map_err(|err| Error::bip32_key_generation_failed(Ed25519KeyPair::KEY_TYPE, err.into()))?;
+
+    let private_key = base_key
+        .derive(&standard_path_to_derivation_path(hd_path))
+        .map_err(|err| Error::bip32_key_generation_failed(Ed25519KeyPair::KEY_TYPE, err.into()))?;
+
+    Ok(private_key)
+}
+
+fn standard_path_to_derivation_path(path: &StandardHDPath) -> DerivationPath {
+    let child_indices = vec![
+        ChildIndex::hardened(path.purpose().as_value().as_number())
+            .expect("Purpose is not Hardened"),
+        ChildIndex::hardened(path.coin_type()).expect("Coin Type is not Hardened"),
+        ChildIndex::hardened(path.account()).expect("Account is not Hardened"),
+        ChildIndex::normal(path.change()).expect("Change is Hardened"),
+        ChildIndex::normal(path.index()).expect("Index is Hardened"),
+    ];
+
+    DerivationPath::new(child_indices)
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Ed25519KeyPair {
+    keypair: Keypair,
+}
+
+impl Clone for Ed25519KeyPair {
+    fn clone(&self) -> Self {
+        let Self {
+            keypair: Keypair { secret, public },
+        } = self;
+        Self {
+            keypair: Keypair {
+                secret: SecretKey::from_bytes(secret.as_bytes()).unwrap(),
```

Comment:
Let's add a comment explaining why this is needed, ie. because `SecretKey` does not implement `Clone`.